browser/src/main/java/org/netbeans/html/presenters/browser/SimpleServer.java

```diff
@@ -0,0 +1,751 @@
+                }
+                bb.clear();
+
+                LOG.log(Level.INFO, "Found page request {0}", url); // NOI18N
+                bb.clear();
+                bb.put("HTTP/1.1 200 OK\r\n".getBytes());
+                bb.put("Connection: close\r\n".getBytes());
+                bb.put("Server: http://dvbcentral.sf.net\r\n".getBytes());
+                bb.put(date(null));
+                bb.put("\r\n".getBytes());
+                bb.put(("Content-Type: " + mime + "\r\n").getBytes());
+                bb.put("Pragma: no-cache\r\nCache-control: no-cache\r\n".getBytes());
+                bb.put("\r\n".getBytes());
+                bb.flip();
+                channel.write(bb);
+                LOG.log(Level.FINER, "Written header, type {0}", mime);
+                bb = null;
+
+                if (justHead) {
+                    LOG.fine("Writer flushed and closed, closing channel");
+                    channel.close();
+                    return;
+                }
+            }
+
+            LOG.log(Level.FINE, "delegating to handler: {0}", h.getClass().getName());
+            try {
+                h.replyTo(header, channel, this);
+                LOG.log(Level.FINE, "replyTo delegated, is channel open: {0}", channel.isOpen());
+            } finally {
+                if (!channel.isOpen()) {
+                    LOG.log(Level.FINE, "channel not open, closing");
+                    key.attach(null);
+                    key.cancel();
+                }
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "DynamicRequest[" + url + ", " + upr.getClass().getName() + "]";
+        }
+
+        public SelectableChannel channel() {
+            return delegate.channel();
+        }
+
+        public Selector selector() {
+            return delegate.selector();
+        }
+
+        public boolean isValid() {
+            return delegate.isValid();
+        }
+
+        public void cancel() {
+            delegate.cancel();
+        }
+
+        public int interestOps() {
+            return delegate.interestOps();
+        }
+
+        public SelectionKey interestOps(int arg0) {
+            return delegate.interestOps(arg0);
+        }
+
+        public int readyOps() {
+            return delegate.readyOps();
+        }
+    } // end of DynamicRequest
+
+    private static final class MsgRequest implements Request {
+
+        private final String msg;
+        private ByteBuffer bb;
+        private int index;
+
+        public MsgRequest(String a) {
+            this.msg = a;
+        }
+
+        public void handle(SelectionKey key, SocketChannel channel) throws IOException {
+            if (bb == null) {
+                bb = ByteBuffer.allocate(8192);
+                bb.put("HTTP/1.1 200 OK\r\n".getBytes());
+                bb.put("Connection: close\r\n".getBytes());
+                bb.put("Server: http://dvbcentral.sf.net\r\n".getBytes());
```

Comment:
What is this? Was this copied from somewhere?

Reply:
Yes, I had already written a simple HTTP server once. I am basing my current work for Apache NetBeans on my prevoius work.

Reply:
I described my problems at [Help [WANTED] by skilled JavaScript gurus](http://mail-archives.apache.org/mod_mbox/netbeans-dev/201911.mbox/%3cCAM6-voGk5Rt+CSaz4viHnQkHRZot4HsG1Gu9j+onYVU4KDZLMw@mail.gmail.com%3e). As I received no help, I decided to eliminate Grizzly.

It is an overkill to use five worker threads and five load balancer threads for a simple local to local communication between browser and JVM. Moreover the inherent multithreading is really hard to debug. The `SimpleServer` shall have just a single thread for all/both connections.

I understand why the @dukescript guys used a ready made solution, but it only complicates the code base. If nothing else, we now have two implementations that can be easily compared side by side - see `ServerFactories` & co.